docs/design/Network Topology Aware Scheduling.md

```diff
@@ -0,0 +1,575 @@
+4. Gang scheduling is also needed in this case, to make sure all pods are able to proceed with their work. 
+
+## Use Case 2
+
+1. LLM training job:  16 NPU per pod, 3000 pod per job
+
+![network topology usecase2](images/network-topology-aware/network-topology-usecase2.png)
+
+2. There are 3 network control planes, which are VPC, Roce and HCCS.  
+3. The pods in Job are expected to be grouped and the pods belonging to the same group have higher demand for the network bandwidth. The tensor parallels are performed on pods in the same group.  
+![network topology usecase3](images/network-topology-aware/network-topology-usecase3.png)
+
+4. The pods belonging to the same group are required to be scheduled to HCCS topology zone.  
+5. Prefer to schedule the pod group to one HCCS topology zone, if not enough nodes, try to schedule to the RoCE topology zone and then the VPC topology zone.  
+6. Gang scheduling is required in this case, to make sure all pods are able to proceed with their work.
+
+# Scope:
+
+In Scope:
+
+* support clos network topology define and management  
+* support network topology aware scheduling for Volcano job  
+* support spine-leaf network
+
+# Function Detail
+
+### network topology management
+
+Option 1: Describe network topology by labels, there's a proposal in upstream  
+Cons: 
+* complicated to construct tree for the topology information  
+
+Option 2: Describe network topology by CRD, NetworkTopology in Volcano → plugin/scheduling
+
+- label \-\> NetworkTopology   
+- Rest API (NV, HW) \-\> NetworkTopology
+
+Pros: 
+  * easier for debugging:  
+  * engineers don’t need to construct the tree of topology info by themselves.   
+  * the scheduler uses the same as engineers can see.
+
+* Components besides scheduler are able to use the CR as well
+
+#### network topology definition
+
+`HyperNode` is a performance domain which consists of a group of nodes or sub-performance domains.The network bandwidth and latency is the same in one HyperNode. This CRD is used to describe the network topology in Kubernetes cluster.
+
+`Tier` is a way to distinguish different performance domains. The bandwidth and latency are the same in one tier. The smaller the value of the tier, the higher the bandwidth. For example, compute-network and storage network can be in different tiers, or in the compute-network, there are several levels of spine, leaf switches, each level can be identified as a tier.
+
+```go
+type HyperNode struct {
+	metav1.TypeMeta `json:”,inline”`
+	metav1.ObjectMeta `json:”metadata, omitempty”`
+
+	Spec HyperNodeSpec `json:”spec”`
+	Status HyperNodeStatus `json:”status”`
+}
+
+type HyperNodeSpec struct {
+	tier string	`json:"tier,omitempty"`
+	members []MemberSpec	`json:"members,omitempty"`
+}
+
+type MemberSpec struct {
+	name string	`json:"name,omitempty"`
+	type string	`json:"type,omitempty"`
```

Comment:
We should capitalize these fields? `type` conflicts with keywords now. Can the `type` field here be enumerated with constants? There are only two types: `Node` and `HyperNode`, right?

Reply:
Here is just an example, and the API definition is below.